### DISCUSSION TITLE
Direction on #4088

### DISCUSSION THREAD

**Discussion by kleenkanteen** [maplibre/maplibre-gl-js#4214] [General]
This is my [first ticket](https://github.com/maplibre/maplibre-gl-js/issues/4088) and I'm new to the codebase. Wondering what approch I should take to debug?

1. Currently I cloned the repo into a github codespace and got the examples in `test/examples` showing.
2. I created a test.html file with the contents of this [jsbin](https://jsbin.com/cahomamazu/edit?html,output) but I replaced the lines pulling the package from unpkg.com to now pulling from `'../../dist/maplibre-gl.css'`. I believe after I make code changes, I should run `npm run build-dev` and reload the test.html file I am looking at to observe the library changes?
3. I looked at `src/ui/control/terrain_control.ts`, [link to it](https://github.com/maplibre/maplibre-gl-js/blob/main/src/ui/control/terrain_control.ts), and found line 58 saying `this._map.setTerrain(this.options)`. So I went into the function `setTerrain()` but there's a lot going on. The word image is not found in the function. What should I look at next to find the interaction it has with an image?

pinging @HarelM @sbachinin

**Comment by HarelM** [comment]
npm run start
Will allow you to see changes as you do them.
Use maplibre-gl-dev.js.

Look for imageSource to dive into how image source work.

There are also the guides in the docs and the readme scattered in this repo.

I would recommend opening PRs to update the docs and readmes if you see some information that could help future newcomers.

**Reply by kleenkanteen** [reply]
@HarelM I looked around in the file and set some print statements to see if the coordinates are changing. Apparently not. Here is a video of me explaining how I found that out: https://streamable.com/8z1t9a. Wondering if I'm on the right track and what I should look into next because the bug seems to be coming from outside the image source file?

Here is a gist of the image_source.ts file with the prints and the testing .html file: https://gist.github.com/kleenkanteen/66ad2ec30850489514253ac1f80a8db1

**Reply by HarelM** [reply]
The coordinates define the location of the image, which doesn't change. 
Inside the image source there's a texture which is drawn into a webgl context. The rest of the manipulation is done using this webgl context.
You'll need to dig into the webgl code to understand what needs to be fixed.

**Reply by kleenkanteen** [reply]
I unassigned myself the ticket. I don't know enough about webgl to do the ticket...

**Reply by HarelM** [reply]
Thanks for the update.
